src/app/core/admin/admin-dashboard/admin-dashboard.component.ts

```diff
@@ -0,0 +1,209 @@
+      const widgetType = event.previousContainer.data[event.previousIndex];
+      const newWidget: DynamicComponentConfig = {
+        component: widgetType.component,
+        config: { ...widgetType.defaultConfig },
+      };
+
+      this.dashboardConfig.widgets.splice(event.currentIndex, 0, newWidget);
+    }
+  }
+
+  async editWidget(widgetConfig: DynamicComponentConfig, idx: number) {
+    const settingsComponent = this.getSettingsComponentforWidget(widgetConfig);
+    if (!settingsComponent) return;
+
+    const updatedConfig = await this.openWidgetSettingsDialog(
+      widgetConfig,
+      settingsComponent,
+    );
+    if (updatedConfig) {
+      this.dashboardConfig.widgets[idx] = updatedConfig;
+      this.dashboardConfig.widgets = [...this.dashboardConfig.widgets];
+    }
+  }
+
+  getSettingsComponentforWidget(widgetConfig: DynamicComponentConfig): any {
+    switch (widgetConfig.component) {
+      case "ShortcutDashboard":
+        return "ShortcutDashboardSettings";
+
+      case "EntityCountDashboard":
+        return "EntityCountDashboardSettings";
+
+      case "ImportantNotesDashboard":
+        return "ImportantNotesDashboardSettings";
+
+      case "TodosDashboard":
+        return "TodosDashboardSettings";
+
+      case "NotesDashboard":
+        return "NotesDashboardSettings";
+
+      case "AttendanceWeekDashboard":
+        return "AttendanceWeekDashboardSettings";
+
+      case "ProgressDashboard":
+        return "ProgressDashboardSettings";
+
+      case "BirthdayDashboard":
+        return "BirthdayDashboardSettings";
+
+      default:
+        return null;
+    }
+  }
+
+  private prettifyWidgetTitle(componentName: string): string {
+    return componentName.replace(/([a-z])([A-Z])/g, "$1 $2");
+  }
+
+  private async openWidgetSettingsDialog(
+    widgetConfig: DynamicComponentConfig,
+    settingsComponent: string,
+  ): Promise<DynamicComponentConfig | undefined> {
+    const dialogData: AdminWidgetDialogData = {
+      widgetConfig: { ...widgetConfig },
+      settingsComponent: settingsComponent,
+      title: `${this.prettifyWidgetTitle(widgetConfig.component)} Settings`,
+    };
+
+    const dialogRef = this.dialog.open(AdminWidgetDialogComponent, {
+      width: "600px",
+      data: dialogData,
+    });
+
+    return firstValueFrom(dialogRef.afterClosed());
+  }
+
+  removeWidget(index: number) {
+    this.dashboardConfig.widgets.splice(index, 1);
+  }
+
+  async addNewWidget() {
+    // Open the widget selection dialog for dashboard widgets
+    const dialogRef = this.dialog.open(WidgetComponentSelectComponent, {
+      width: "400px",
+      data: { isDashboard: true },
+    });
+
+    const selectedWidget = await firstValueFrom(dialogRef.afterClosed());
+    if (selectedWidget) {
+      this.dashboardConfig.widgets.push(selectedWidget);
+    }
+  }
+
+  cancel() {
+    this.dashboardConfig = JSON.parse(
+      JSON.stringify(this.originalDashboardConfig),
+    );
```

Comment:
can you test if we need this? I think we navigate away and thereby discard the whole component instance, so it should not make a difference if we update a class property (`this.dashboardConfig`) or not.

(As we don't save the config to the DB, changes should automatically be lost anyway)

Reply:
If we can remove it here, I think we don't need `this.originalDashboardConfig` at all and can delete that property completely.